application/src/main/java/org/thingsboard/server/controller/BaseController.java

```diff
@@ -378,15 +376,11 @@ private ThingsboardException handleException(Exception exception, boolean logExc
         } else if (exception instanceof AsyncRequestTimeoutException) {
             return new ThingsboardException("Request timeout", ThingsboardErrorCode.GENERAL);
         } else if (exception instanceof DataAccessException) {
-            Throwable rootCause = ExceptionUtils.getRootCause(exception);
-            if (rootCause instanceof PSQLException) {
-                String sqlError = Optional.ofNullable(((PSQLException) rootCause).getServerErrorMessage())
-                        .map(ServerErrorMessage::getMessage).orElse(null);
-                if (isNotEmpty(sqlError)) {
-                    sqlError = StringUtils.capitalize(sqlError);
-                    return new ThingsboardException(sqlError, ThingsboardErrorCode.GENERAL);
-                }
+            String errorType = exception.getClass().getSimpleName();
+            if (!logControllerErrorStackTrace) { // not to log the error twice
+                log.warn("Database error: {} - {}", errorType, ExceptionUtils.getRootCauseMessage(exception));
             }
+            return new ThingsboardException("Database error: " + errorType, ThingsboardErrorCode.GENERAL);
```

Comment:
Reporting an error type may also explain how an attacker query affects the DB. e.g., looking for a security flow expecting something like SQLSyntaxException